components/telemetry-operator/internal/configbuilder/config_builder.go

```diff
@@ -0,0 +1,82 @@
+package configbuilder
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+)
+
+type PipelineConfig struct {
+	InputTag          string
+	MemoryBufferLimit string
+	StorageType       string
+	FsBufferLimit     string
+}
+
+// MergeSectionsConfig merges Fluent Bit filters and outputs to a single Fluent Bit configuration.
+func MergeSectionsConfig(pipeline *telemetryv1alpha1.LogPipeline, pipelineConfig PipelineConfig) (string, error) {
+	var sb strings.Builder
+	sb.WriteString(createRewriteTagFilterSection(pipeline, pipelineConfig))
+	sb.WriteString(createRecordModifierFilter(pipeline))
+	sb.WriteString(createCustomFilters(pipeline))
+	sb.WriteString(createKubernetesMetadataFilter(pipeline))
+	sb.WriteString(createLuaDedotFilter(pipeline))
+	sb.WriteString(createOutputSection(pipeline, pipelineConfig))
+
+	return sb.String(), nil
```

Comment:
This function always returns nil as error.

Reply:
Yes, was on my list and I forgot 😬  Just pushed an update on this, where validation on the custom multiline is done up front. We can decide on, where this has to be done though.

Reply:
Please have look on this again @skhalash @dennis-ge